tests/test_switch.py

```diff
@@ -777,8 +778,8 @@ async def turn_light(state, **kwargs):
     assert (
         switch.extra_state_attributes["autoreset_time_remaining"][light.entity_id] > 0
     )
-    await asyncio.sleep(0.3)  # Should be enough time for auto reset
     await update()
+    await asyncio.sleep(0.3)  # Should be enough time for auto reset
```

Comment:
Why did you move this statement?

Reply:
This fixes the broken test (what I tried to explain in https://github.com/basnijholt/adaptive-lighting/pull/598#issuecomment-1566170628).

Reply:
This bit is the only reason I haven't clicked merge yet. I want to understand why this is required and I am wondering whether this indicates whether anything broke.

Reply:
I think the reason this is required is that the service call is now executed from the event loop (due to now being `async`), not directly/synchronously as it used to be. So I could have kept the `sleep(0.3)` where it was and added another `sleep` here to await processing of the event loop. Simply moving it down seemed simpler. But like I said, I'm not entirely sure because I'm far from being a Python expert :)

Reply:
So what's the status, did my answer help in any way? Whether you merge this or not is both fine with me, but it would be helpful if I knew what to expect.

Reply:
I’ll just go on good faith and merge 😃

Thanks a lot for your contribution!